src/helpers/permissionHelpers.ts

```diff
@@ -13,9 +30,28 @@ export function getRequiredPermissions(): bigint[] {
 }
 
 export function memberIsModerator(member: GuildMember): boolean {
-	return member.permissions.has(Permissions.FLAGS.KICK_MEMBERS);
+	return member.permissions.has(Permissions.FLAGS.KICK_MEMBERS, true);
 }
 
 export function memberIsAdmin(member: GuildMember): boolean {
 	return member.permissions.has(Permissions.FLAGS.ADMINISTRATOR);
 }
+
+// Fixes https://github.com/MarcusOtter/discord-needle/issues/23
+// Should not be required, but Discord for some reason allows the default duration to be higher than the allowed value
+export function getSafeDefaultAutoArchiveDuration(channel: TextChannel | NewsChannel): ThreadAutoArchiveDuration {
+	const archiveDuration = channel.defaultAutoArchiveDuration;
+	if (!archiveDuration || archiveDuration === "MAX") return "MAX";
+
+	const highest = getHighestAllowedArchiveDuration(channel);
+	return archiveDuration > highest
+		? highest
+		: archiveDuration;
+}
+
+function getHighestAllowedArchiveDuration(channel: TextChannel | NewsChannel) {
+	if (channel.guild.features.includes("SEVEN_DAY_THREAD_ARCHIVE")) return 10080;
+	if (channel.guild.features.includes("THREE_DAY_THREAD_ARCHIVE")) return 4320;
+
+	return 1440; // 1d
+}
```

Comment:
Is there an Enum available for this? There has to be in djs, right?

Reply:
There is [ThreadAutoArchiveDuration](https://discord.js.org/#/docs/main/stable/typedef/ThreadAutoArchiveDuration) but it's not an enum, just a type. I couldn't find one that works anyways

![image](https://user-images.githubusercontent.com/35617441/152166430-d2caf409-8c05-491a-9756-c4342e995c5c.png)

Reply:
What a shame, djs.